R/bm-write-csv.R

```diff
@@ -23,12 +25,20 @@ write_csv <- Benchmark(
     BenchEnvironment(
       write_csv_func = get_csv_writer(writer),
       source = source,
-      df = df
+      df = df,
+      compression = compression
     )
   },
   # delete the results before each iteration
   before_each = {
     result_file <- tempfile(fileext = ".csv")
+    if (compression != "uncompressed") {
+      if (compression == "gzip") {
+        result_file <- paste0(result_file, ".gz")
+      } else {
+        result_file <- paste0(result_file, ".", compression)
+      }
+    }
```

Comment:
Could we do this in the `setup` block? That would match some of our other setups